### PR TITLE
BC breaks for v2.0

### PR DESCRIPTION
**Pull request by milo**
There are proposes for BC breaks :o) Every commit references own issue. I'll leave few comments among diffs.

### CODE REVIEW

File: src/Framework/TestCase.php [hunk 1]
```diff
@@ -29,35 +29,31 @@ class TestCase
...
-		}, $r->getMethods())));
-
-		if (substr($method, 0, 2) === '--') { // back compatibility
-			$method = NULL;
-		}
+		}, (new \ReflectionObject($this))->getMethods())));
 
-		if ($method === NULL && isset($_SERVER['argv']) && ($tmp = preg_filter('#(--method=)?([\w-]+)$#Ai', '$2', $_SERVER['argv']))) {
+		# TODO: Support 'php test.phpt testMethod' or 'php test.phpt --method=testMethod' only?
+		if (isset($_SERVER['argv']) && ($tmp = preg_filter('#(--method=)?([\w-]+)$#Ai', '$2', $_SERVER['argv']))) {
```
Comment (milo) [inline src/Framework/TestCase.php]: Not sure about support manual args without `--method=`.

File: src/Framework/TestCase.php [hunk 2]
```diff
@@ -70,6 +66,12 @@ public function run($method = NULL)
 	 */
 	public function runTest($method, array $args = NULL)
 	{
+		if (!method_exists($this, $method)) {
+			throw new TestCaseException("Method '$method' does not exist.");
```
Comment (milo) [inline src/Framework/TestCase.php]: Actually this is not needed. The `ReflectionException` will be thrown. But this have a straightforward message.

File: src/Framework/TestCase.php [hunk 3]
```diff
@@ -70,6 +66,12 @@ public function run($method = NULL)
...
+		} elseif (!preg_match(self::METHOD_PATTERN, $method)) {
+			throw new TestCaseException("Method '$method' is not a testing method.");
```
Comment (milo) [inline src/Framework/TestCase.php]: This forces `runTest()` usage only for test methods. IMHO OK, any other public method can be called directly. But maybe, there can be some use case for it.

**Comment by milo** [comment]
Output may look like:
```
$ src/tester tests
 _____ ___  ___ _____ ___  ___
|_   _/ __)( __/_   _/ __)| _ )
  |_| \___ /___) |_| \___ |_|_\  v2.0.x

Note: Default interpreter is CLI since Tester v2.0. It used to be CGI.
Note: No php.ini is used.
PHP 5.6.13 (cli) | php -n  | 8 threads

-- FAILED: tests\Framework\TestCase.annotationThrows.phpt
   Exited with error code 255 (expected 0)
   LogicException: Calling TestCase::run($method) is deprecated. Use TestCase::runTest($method).
```

**Comment by dg** [comment]
:+1:

File: src/Framework/TestCase.php [hunk 4]
```diff
@@ -29,35 +29,31 @@ class TestCase
...
+		}
+
 		$methods = array_values(preg_grep(self::METHOD_PATTERN, array_map(function (\ReflectionMethod $rm) {
 			return $rm->getName();
-		}, $r->getMethods())));
-
-		if (substr($method, 0, 2) === '--') { // back compatibility
-			$method = NULL;
-		}
+		}, (new \ReflectionObject($this))->getMethods())));
```
Comment (JanTvrdik) [inline src/Framework/TestCase.php]: Extract to `private getTestMethods()`?

File: src/Framework/TestCase.php (hunk 1 above)
Comment (dg) [inline src/Framework/TestCase.php]: I am not using TestCase, but I think that it is not needed to have two ways how do the same think and it blocks first parameter.

File: src/Framework/TestCase.php (hunk 4 above)
Comment (milo) [inline src/Framework/TestCase.php]: IMHO not needed. It is used only once.

**Pull request merged**